### ISSUE TITLE
Missing Translation

### ISSUE THREAD

**Issue by gregBerthelot** [KIProtect/klaro#11]
Hi

First thanks for this great job. I am running it locally without any issue. Unfortunately I have an issue in live on our own CMS. On notice I get [missing translation: ,purposes/[object Set],/,en,]

The rest works fine. Any idea what can be the cause?

Thanks
Greg

**Comment by adewes** [comment]
Thanks for reporting this, seems like a missing translation for a purpose (or a wrong data type for the `purposes` object). Can you share your `config.js` or post a link to the site? I'll gladly have a look at it then!
In general, make sure that `purposes` is an array, e.g. ` purposes: ['analytics']`. When I change the purpose to something unknown I get an error like `[missing translation: ,purposes/securisty,/,en,]`, so it might really be a problem with the data type.

**Comment by gregBerthelot** [comment]
Thanks for reply. Curious thing is that the same code works locally and I am unable to reproduce the issue. Seems that I have a conflict on the CMS. Will put site live and share link.

**Comment by adewes** [comment]
Thanks for posting the link @gregBerthelot ! Concerning the problem with the translations it seems that you might be using an old version of Klaro (we will add proper versioning and releases soon btw), could you try to download the new version from Master and try again with that?

**Comment by gregBerthelot** [comment]
Thanks but unfortunately that does not work better.

**Comment by gregBerthelot** [comment]
HI had the need to publish the website without the issue. So I have remove the `{purposes}` on it. The issue is visible on this [test website](https://cms.ipaoo.fr/testCookies) to investigate

**Comment by adewes** [comment]
So I did some debugging to see if this issue is related to Klaro and found the problem:
* I made a local copy of your site and successively disabled all scripts to see if one of them caused the behavior
* When disabling `fat_eau.js` (which is probably not active / included in the offline version) the problem disappears.
* I unminified the source of the module and looked through anything that could interfere with the `getPurposes` function from Klaro (which constructs a list of purposes from your app config and apparently did not return an Array but instead a set, which in turn caused the translation function to fail)
* Sure enough, `fat_eau.js` overwrites (and breaks) the `from` method of the `Array` class:

```
    Array.from = function(item) {
        if (item == null) return [];
        return (Type.isEnumerable(item) && typeof item != 'string') ? (typeOf(item) == 'array') ? item : slice.call(item) : [item];
    };
```
I suggest you fix the problem in your library (if you have control over it) or exclude it by default, as overwriting well-defined behavior is a bad practice in general and will often lead to hard-to-reproduce bugs.
Hope this helps, I'm closing this issue now as it's not related to Klaro, let me know if you have any questions!
